probe-rs/src/probe/jlink/mod.rs

```diff
@@ -803,30 +802,19 @@ impl JLink {
         let max_bits = (self.max_mem_block_size - COMMAND_OVERHEAD) / 2 * 8;
         let max_bits = std::cmp::min(max_bits, 65535);
 
-        let dir_chunks = dir.into_iter().chunks(max_bits);
         let swdio_chunks = swdio.into_iter().chunks(max_bits);
 
-        #[allow(clippy::useless_conversion)]
-        let chunks = dir_chunks.into_iter().zip(swdio_chunks.into_iter());
-
         let mut output = Vec::new();
         let mut buf = Vec::with_capacity(self.max_mem_block_size);
         buf.resize(COMMAND_OVERHEAD, 0);
 
-        for (dir, swdio) in chunks {
+        for swdio in swdio_chunks.into_iter() {
```

Comment:
Just by reading the change, it's unclear to me what is happening. `swdio` is a chunk of `IoSequenceItem`s, but then it's treated as a slice of u8s? bools?. I think something around this may not happen exactly as expected.

Reply:
Nothing has changed in this division comparing to master. Here, it is still just dividing data into chunks. Not type conversion happens here. 

The problem is few lines below where I forgot to adjust my solution and replace this line: 
`buf.extend(dir.inspect(|_| dir_bit_count += 1).collapse_bytes());`

This part of implementation is not written to express clearly the intention as one would whish for. I don't know the J-LInk details so I got lost easily. Maybe something to improve for the future. 

I will try to fix it tomorrow. Thanks @bugadani for the support. Good that tests caught that. I was able to reproduce de failure in my case by grabbing some j-link and trying to connect some STM board.

Reply:
I just recreated (dir,bit) iterators as it was used before my change.